### PR TITLE
479: Uninstall routine

### LINKED ISSUE #479: Provide an uninstall routine
GlotPress should use `uninstall.php` ~and/or have a wp-cli command~ to remove/reset all options and tables.

### PR DESCRIPTION
**Pull request by toolstack**
Resolves #479.

### CODE REVIEW

File: uninstall.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	// Include the schema so we can get the list of tables.
+	if ( ! function_exists( 'gp_schema_get' ) ) {
+		include( GP_PATH . GP_INC . 'schema.php' );
+	}
+
+	// Get the schema and loop through each table so we can delete it.
+	$schema = gp_schema_get();
+
+	foreach ( $schema as $table => $sql ) {
+		$table_name = $wpdb->prefix . 'gp_' . $table;
```
Comment (ocean90) [inline uninstall.php]: This looks wrong, the prefix can be customized via `$gp_table_prefix`.

File: uninstall.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+		// We can't use $wpdb->prepare here as the table name is not one of the support data types.
+		$wpdb->query( "DROP TABLE {$table_name};" ); // WPCS: unprepared SQL ok.
+	}
+
+	// Delete the WordPress options we use.
```
Comment (ocean90) [inline uninstall.php:52]: 'WordPress' sounds confusing in this context, the options are used by GlotPress.
Better: `// Delete options.`

There is a lot of 'we' in this function too, this should be avoided.

File: uninstall.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	// Include the schema to get the list of tables.
+	if ( ! function_exists( 'gp_schema_get' ) ) {
+		include( GP_PATH . GP_INC . 'schema.php' );
+	}
+
+	// Get the schema and loop through each table to delete it.
+	$schema = gp_schema_get();
+
+	foreach ( $schema as $table => $sql ) {
+		$table_name = $wpdb->{'gp_' . $table};
```
Comment (ocean90) [inline uninstall.php]: Does this work if the plugin is disabled?

Comment (toolstack) [reply uninstall.php]: Yes, that's why the additional defines are included above.

File: gp-includes/gp.php [hunk 4]
```diff
@@ -165,4 +165,13 @@ class GP {
...
 	public static $scripts = array();
+
+	/**
+	 * Model for GP_WP_Settings.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @var GP_WP_Settings
+	 */
+	public static $wp_settings;
```
Comment (ocean90) [inline gp-includes/gp.php]: This looks weird. Can we make that `$admin` and use it for all (future) stuff which is handled in the WordPress admin?

This should also only be initialized when we're in the admin, not on the front end.

**Comment by ocean90** [comment]
Closing in favor of #1247.

**Pull request closed without merging**